x/data/client/testsuite/grpc.go

```diff
@@ -159,6 +161,249 @@ func (s *IntegrationTestSuite) TestQueryHashByIRI() {
 	}
 }
 
+func (s *IntegrationTestSuite) TestIRIByRawHash() {
+	val := s.network.Validators[0]
+
+	iri, ch := s.createIRIAndRawHash([]byte("xyzabc123"))
+
+	encodedHash := encodeBase64Bytes(ch.Raw.Hash)
+
+	testCases := []struct {
+		name   string
+		url    string
+		expErr bool
+		errMsg string
+	}{
+		{
+			"empty hash",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-raw?digest_algorithm=%s",
+				val.APIAddress,
+				ch.Raw.DigestAlgorithm, // enum 1
+			),
+			true,
+			"hash cannot be empty",
+		},
+		{
+			"invalid hash",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-raw?hash=%s&digest_algorithm=%s",
+				val.APIAddress,
+				"foo",
+				ch.Raw.DigestAlgorithm, // enum 1
+			),
+			true,
+			"failed to decode base64 string",
+		},
+		{
+			"unspecified digest algorithm",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-raw?hash=%s",
+				val.APIAddress,
+				encodedHash, // base64 encoded string
+			),
+			true,
+			"digest algorithm cannot be unspecified",
+		},
+		{
+			"invalid digest algorithm",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-raw?hash=%s&digest_algorithm=%s",
+				val.APIAddress,
+				encodedHash, // base64 encoded string
+				"foo",
+			),
+			true,
+			"foo is not a valid data.DigestAlgorithm",
+		},
+		{
+			"invalid media type",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-raw?hash=%s&digest_algorithm=%d&media_type=%s",
+				val.APIAddress,
+				encodedHash,            // base64 encoded string
+				ch.Raw.DigestAlgorithm, // enum 1
+				"foo",
+			),
+			true,
+			"foo is not a valid data.RawMediaType",
+		},
+		{
+			"valid request",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-raw?hash=%s&digest_algorithm=%d&media_type=%d",
+				val.APIAddress,
+				encodedHash,            // base64 encoded string
+				ch.Raw.DigestAlgorithm, // enum 1
+				ch.Raw.MediaType,       // enum 0
+			),
+			false,
+			"",
+		},
+		{
+			"valid request enums as strings",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-raw?hash=%s&digest_algorithm=%s&media_type=%s",
+				val.APIAddress,
+				encodedHash,                    // base64 encoded string
+				"DIGEST_ALGORITHM_BLAKE2B_256", // enum 1
+				"RAW_MEDIA_TYPE_UNSPECIFIED",   // enum 1
+			),
+			false,
+			"",
+		},
+	}
+
+	require := s.Require()
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			bz, err := rest.GetRequest(tc.url)
+			require.NoError(err)
+
+			var res data.QueryIRIByRawHashResponse
+			err = val.ClientCtx.Codec.UnmarshalJSON(bz, &res)
+
+			if tc.expErr {
+				require.Error(err)
+				require.Contains(string(bz), tc.errMsg)
```

Comment:
should be able to use `ErrorContains` here

Reply:
`bz` is the response from `GetRequest` so we are checking if the error message is within the get response. The `err` above is from `UnmarshalJSON` so we need to check if the response contains the error message here.